src/service/user/UserService.ts

```diff
@@ -19,5 +19,28 @@ export const userService = () => {
       throw new HttpError('Errors validating the user:' + errors, HttpStatusCode.BAD_REQUEST)
     }
   }
-  return { createUserService }
-}
+  const editUser = async ({ id, name, email, telephone, type }) => {
+    const userRepository = getRepository(User)
+    const user = await userRepository.findOne(id)
+    if (!user) {
+      throw new HttpError('User not found with: ' + id, HttpStatusCode.BAD_REQUEST)
```

Comment:
aqui acredito que o melhor http status code seria um not found (404), pois o usuário não foi encontrado 
https://http.cat/404